src/com/cft/contactmerge/tests/CompareContactPartsTest.java

```diff
@@ -5,67 +5,302 @@
+    }
+    //---------------------------------------------------------------------------- Basic Names Tests ----------------------------------------------------------------------------
     @Test
     void doNamesMatch_No_DifferentLastName() {
-        assertEquals(AnswerType.no, CompareContactParts.doNamesMatch("John", "Doe", "John", "Adams"));
+        assertEquals(AnswerType.no, CompareContactParts.doNamesMatch("John", "Doe", "John", "Adams"), firstAndLastNameFailedMsg("John", "Doe", "John", "Adams"));
     }
 
     @Test
     void doNamesMatch_No_DifferentFirstName() {
-        assertEquals(AnswerType.no, CompareContactParts.doNamesMatch("John", "Doe", "Jane", "Doe"));
+        assertEquals(AnswerType.no, CompareContactParts.doNamesMatch("John", "Doe", "Jane", "Doe"), firstAndLastNameFailedMsg("John", "Doe", "Jane", "Doe"));
     }
 
     @Test
     void doNamesMatch_Yes_CaseDoesNotMatter() {
-        assertEquals(AnswerType.yes, CompareContactParts.doNamesMatch("john", "DOE", "JOHN", "Doe"));
+        assertEquals(AnswerType.yes, CompareContactParts.doNamesMatch("john", "DOE", "JOHN", "Doe"), firstAndLastNameFailedMsg("john", "DOE", "JOHN", "Doe"));
     }
 
     @Test
     void doNamesMatch_Yes_IgnoreSpaces() {
-        assertEquals(AnswerType.yes, CompareContactParts.doNamesMatch(" John", "Doe", "John ", " Doe  "));
+        assertEquals(AnswerType.yes, CompareContactParts.doNamesMatch(" John", "Doe", "John ", " Doe  "), firstAndLastNameFailedMsg(" John", "Doe", "John ", " Doe  "));
+    }
+
+    @Test
+    void doNamesMatch_Yes_IgnoreSpaces_CaseInsensitive() {
+        assertEquals(AnswerType.yes, CompareContactParts.doNamesMatch(" JoHn", "DOE", "JOHN ", " Doe  "), firstAndLastNameFailedMsg(" JoHn", "DOE", "JOHN ", " Doe  "));
+    }
+
+    //------------------------------------------------------------------------- Punctuation First Names -------------------------------------------------------------------------
+    @Test
+    void doNamesMatch_Yes_IgnorePunctuationInFirstName_CaseInsensitive_IgnoreSpaces() {
+        assertEquals(AnswerType.yes, CompareContactParts.doNamesMatch("aN-denIse ", "Doe ", " An Denise ", "   doe "), firstAndLastNameFailedMsg("aN-denIse ", "Doe ", " An Denise ", "   doe "));
+    }
+
+    @Test
+    void doNamesMatch_Yes_IgnoreMultiPunctuationInFirstName_CaseInsensitive_IgnoreSpaces() {
+        assertEquals(AnswerType.yes, CompareContactParts.doNamesMatch("aN-De-denIse ", "dOe ", " An dE Denise ", "   doe "), firstAndLastNameFailedMsg("aN-De-denIse ", "dOe ", " An dE Denise ", "   doe "));
+    }
+
+    @Test
+    void doNamesMatch_No_IgnoreMultiPunctuationInFirstName_CaseInsensitive_IgnoreSpaces_DifferentLastNames() {
+        assertEquals(AnswerType.no, CompareContactParts.doNamesMatch(" aN-De-denIse", " dOe", "   An dE Denise  ", " adams  "), firstAndLastNameFailedMsg(" aN-De-denIse", " dOe", "   An dE Denise  ", " adams  "));
```

Comment:
This test doesn't seem to add anything. We already have a test for hyphens in a name and a test for different last names.

Reply:
I was thinking this would help with handling priority testing...
(firstname == yes) and (lastname == yes) == yes
(firstname == yes) and (lastname == maybe) == maybe
(firstname == yes) and (lastname == no) == no
(firstname == maybe) and (lastname == yes) == maybe
(firstname == maybe) and (lastname == maybe) == maybe
(firstname == maybe) and (lastname == no) == no
(firstname == no) and (lastname == maybe) == no
(firstname == no) and (lastname == no) == no

Let me know what you think...